fe/fe-core/src/main/java/com/starrocks/journal/JournalWriter.java

```diff
@@ -0,0 +1,122 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.journal;
+
+import com.starrocks.common.Config;
+import com.starrocks.common.util.Daemon;
+import com.starrocks.metric.MetricRepo;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mortbay.log.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * An independent thread to write journals by batch asynchronously.
+ * Each thread that needs to write a log can put the log in a blocking queue, while JournalWriter constantly gets as
+ * many logs as possible from the queue and write them all to BDB in one transaction.
+ * After committing, JournalWriter will notify the caller thread for consistency.
+ */
+public class JournalWriter extends Daemon {
+    public static final Logger LOG = LogManager.getLogger(JournalWriter.class);
+    // other threads can put log to this queue by calling Editlog.logEdit()
+    private BlockingQueue<JournalTask> logQueue;
+    private Journal journal;
+
+    // store stagging log before commit
+    protected List<JournalTask> staggingLogs = new ArrayList<>();
+    // used for checking if edit log need to roll
+    protected long rollEditCounter = 0;
+
+    public JournalWriter(Journal journal, BlockingQueue<JournalTask> logQueue) {
+        super("JournalWriter", 0L);
+        this.journal = journal;
+        this.logQueue = logQueue;
+    }
+
+    @Override
+    protected void runOneCycle() {
+        try {
+            // 1. waiting if necessary until an element becomes available
+            JournalTask task = logQueue.take();
+
+            long start = System.nanoTime();
+            long uncommitedEstimatedBytes = 0;
+            staggingLogs.clear();
+
+            // 2. batch write
+            while (true) {
+                if (!staggingLogs.isEmpty()) {
+                    task = logQueue.take();
+                }
+                // 2.1 write(no commit)
+                journal.writeWithinTxn(task.getOp(), task.getBuffer());
```

Comment:
why not write code like 
```
try {
} finally {
     journal.commit()
}
```
This can make sure that the current transaction can be commit or abort